hey-discount/src/main/java/Ad.java

```diff
@@ -0,0 +1,177 @@
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+class Review {
+    private String username;
+    private Date date;
+    private String reviewText;
+
+    public Review() {
+        username = "";
```

Comment:
It is a good practice you have multiple overloads of the same method, to reduce the number of repeating code to the minimum. Therefore same-named methods are ordered from the one with the least arguments to the one with the most args and call each other. Constructors are no different. Therefore, your first constructor can look like 
```
public Review() {
    this("", new Date(), "");
}
```
... so that private fields are only assigned in one particular place
... and remember do declare constants for all literals and "magic numbers"!